BiweeklyMeetings/2022-07-05.md

```diff
@@ -0,0 +1,274 @@
+
+**Matthias** - :  So you mean that, ah, it's captured in the policy itself.
+
+**Sebastien** -:  Yeah. Exactly, yeah.
+
+**Matthias** - :  Yeah? Okay, okay. I guess then you could make the argument then.
+
+**Sebastien** -:  I think this would solve this issue. So I feel like we could revisit this if we just come up with the solution for the logging problem, which requires us to come up with a sip for how to come handle topics for log statements, which doesn't have right now. And then another sip for how we want Cardano to change to make this efficient in some way, shape or form.
+
+**Matthias** - :  Yeah, yeah, yeah. Because the logging, we have it at a stage, but you only have it, I think if the script is failing. So the script is happily successful, I'm not sure you can get any trace really out of it. But yeah, I guess the first good step is a CIP, which might requires another CIP, which would be a new product built in for doing this topic centric logging.
+
+**Sebastien** -:  Yeah. Which I think is the right solution because we could try matching a topic into the existing logging scheme, but it would be ugly. So probably it's better just come up with a new built in, come up with a solution to make it efficiently, and then we need some hassles [inaudible 00:10:52] developer to volunteer to do it, which is obviously the hardest part.
+
+**Matthias** - :  Yeah. I think you're going to get a lot of pushback on the logging idea because as I was saying, Plutus is just one big expression right now, there is no statement or really positive execution. So, when do you put a statement and where do you evaluate a statement is up to the interpreter at this stage, even though everything is tricked and it's pretty much, well, intuitive. That's probably getting the way of Bitwise optimizations and stuff like that. If you start requiring stuff to be done in sequence, and to be able to rely on that as a contract writer, that might be tricky. Perhaps actually, Les also has some inputs on that because from working with Plutus and all that stuff, that's maybe something you also thought of. I'm not sure if you want to.
+
+**Sebastien** -:  Yeah, it don't necessarily mean to hijack the CIP meeting to discuss this, but definitely Les, if you have some thoughts on this, or if you wanted to take this offline after this call and discuss this, I'd be really interested because this is something that we want for a few different projects on our side, and I'm sure this would unlock some other stack or other use cases like this token metadata on chain registry.
+
+**Matthias** - :  So sorry, I'm just writing on the chat. So yeah, let's just make a CIP offline discussions out of it for now and see. But yeah, I think my sentiment is that you would probably get some pushback from the Plutus core team because you're going to [inaudible 00:12:47].
```

Comment:
```suggestion
**Matthias** - :  So sorry, I'm just writing on the chat. So yeah, let's just make a CIP offline discussions out of it for now and see. But yeah, I think my sentiment is that you would probably get some pushback from the Plutus core team because you're going to introduce side-effects.